packages/app-page-builder/src/render/plugins/elements/button/Button.tsx

```diff
@@ -32,23 +49,41 @@ const Button = ({ element }: { element: PbElement }) => {
             {({ getAllClasses, elementStyle, elementAttributes }) => {
                 // Use per-device style
                 const justifyContent = elementStyle[`--${kebabCase(displayMode)}-justify-content`];
-
                 return (
-                    <div style={{ display: "flex", justifyContent }}>
-                        <Link
-                            to={link.href || "/"}
-                            target={link.newTab ? "_blank" : "_self"}
-                            style={
-                                !link.href
-                                    ? { ...elementStyle, pointerEvents: "none" }
-                                    : elementStyle
-                            }
-                            {...elementAttributes}
-                            className={getAllClasses(...classes)}
-                        >
-                            {content}
-                        </Link>
-                    </div>
+                    <>
+                        {action.actionType === "onClickHandler" ? (
+                            <div
+                                style={{ display: "flex", justifyContent }}
+                                onClick={
```

Comment:
Instead of creating the function on each render here, let's move this above, and create the callback via `useCallback` hook. We could do it below the `if (link && !action) { ... } else { ...}` block, maybe line 33.